scene/2d/tile_map_layer.cpp

```diff
@@ -1614,6 +1616,68 @@ void TileMapLayer::_update_notify_local_transform() {
 	set_notify_local_transform(notify);
 }
 
+void TileMapLayer::_queue_internal_update() {
+	if (pending_update) {
+		return;
+	}
+	if (is_inside_tree()) {
+		pending_update = true;
+		callable_mp(this, &TileMapLayer::_deferred_internal_update).call_deferred();
+	}
```

Comment:
Could move this to `if (pending_update) {` as `if (pending_update || !is_inside_tree()) {`

Reply:
I think it is clear enough that way. I would probably add a comment about why we don't update while not being in the tree. Basically: 1) it's useless and 2) it's creating issues.